myapp/models.py

```diff
@@ -0,0 +1,242 @@
+from datetime import datetime
+from flask.ext.login import UserMixin, AnonymousUserMixin
+from myapp import db, login_manager
+
+# Association tables for Many-To-Many relationships between various tables
+association_table_class_user = db.Table('association_class_user',
+                                        db.Column('class_id', db.String(64),
+                                                  db.ForeignKey('Class.id')),
+                                        db.Column('user_id', db.String(64),
+                                                  db.ForeignKey('User.id')))
+association_table_class_lab = db.Table('association_class_lab',
+                                       db.Column('class_id', db.String(64),
+                                                 db.ForeignKey('Class.id')),
+                                       db.Column('lab_id', db.String(64),
+                                                 db.ForeignKey(
+                                                    'Lab_info.lab_id')))
+association_table_user_lab = db.Table('association_user_lab',
+                                      db.Column('user_id', db.String(64),
+                                                db.ForeignKey('User.id')),
+                                      db.Column('lab_id', db.String(64),
+                                                db.ForeignKey(
+                                                    'Lab_info.lab_id')))
+
+
+# set up class to track the date and time information for an object
+class DateTimeMixin(object):
+    created_on = db.Column(db.DateTime, default=datetime.now)
+    updated_on = db.Column(db.DateTime, default=datetime.now,
+                           onupdate=datetime.now)
+
+
+# Store general info of a lab
+class Lab_info(db.Model):
```

Comment:
I'm not going to review this PR until it follows pep8. For reference, this violates: https://www.python.org/dev/peps/pep-0008/#naming-conventions

My reasoning is basically that following the style guide is the most basic indication that the author has considered their readers, that they keep in mind the difficulty of reviewing other people's code, and have made an effort to make it legible and easy to understand. It's important that the code does what it's supposed to. It's almost as important that the code do it in an unsurprising, easy to follow manner, because it will be read many many more times than it is written.